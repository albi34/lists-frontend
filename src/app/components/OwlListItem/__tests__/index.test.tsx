import React from 'react';
import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { OwlListItem } from '../index';

const renderItem = (store: Store) => {
  return render(
    <Provider store={store}>
      <OwlListItem index={0} key="0" text="test" status="pending" readOnly />
    </Provider>,
  );
};

describe('<ListItem />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match snapshot', () => {
    const ListItem = renderItem(store);
    expect(ListItem.container.firstChild).toMatchSnapshot();
  });
});
