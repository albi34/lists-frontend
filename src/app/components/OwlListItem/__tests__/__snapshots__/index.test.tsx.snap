// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should match snapshot 1`] = `
<div>
  <input
    readonly=""
    type="text"
    value="test - Pending upload"
  />
  <button>
    X
  </button>
  <button>
    ↑
  </button>
  <button>
    ↓
  </button>
</div>
`;
